File: sol/solution/0001-0100/0007. Reverse Integer/Solution.php
class Solution {
    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        $isNegative = $x < 0;
    $x = abs($x);
    $rev = (int) strrev((string)$x);
    if ($isNegative) {
        $rev = -$rev;
    }
    if ($rev < -2147483648 || $rev > 2147483647) {
        return 0;
    }
    return $rev;
    }
}

File: sol/solution/0001-0100/0007. Reverse Integer/Solution.py
class Solution(object):
    def reverse(self, x):
        INT_MAX, INT_MIN = 2**31 - 1, -2**31
        sign = -1 if x < 0 else 1
        x = abs(x)
        rev = 0
        while x != 0:
            rev = rev * 10 + x % 10
            x //= 10
        rev *= sign
        return rev if INT_MIN <= rev <= INT_MAX else 0